fluent/fluent.go

```diff
@@ -559,6 +559,41 @@ func (n *nextHopEntry) WithIPAddress(addr string) *nextHopEntry {
 	return n
 }
 
+// WithInterfaceRef specifies an interface to be used for the next-hop.
+func (n *nextHopEntry) WithInterfaceRef(name string) *nextHopEntry {
+	n.pb.NextHop.InterfaceRef = &aftpb.Afts_NextHop_InterfaceRef{
+		Interface: &wpb.StringValue{Value: name},
+	}
+	return n
+}
+
+// WithInterfaceSubinterfaceRef specifies both an interface and a
+// specific subinterface to be used for the next-hop.
+func (n *nextHopEntry) WithInterfaceSubinterfaceRef(name string, sub uint64) *nextHopEntry {
```

Comment:
I'm struggling a bit with the naming here -- `WithInterface` is reasonable, since this is going to be applicable for L2 next-hops. I wonder whether just `WithSubInterface(interface string, subinterface uint32)` is reasonable, since a subinterface is never valid by itself. WDYT?

Reply:
Done. (First argument still just `name`, since `interface` is a reserved word)